### PR TITLE
Add new "skip-private" option

### PR DESCRIPTION
**Pull request by igrmk**
Really we never need JSON tags for private fields
So this is such an option

For example you may have following struct

    type Struct struct {
        Public  int
        private int
    }

Currently gomodifytags would add these tags

    type Struct struct {
        Public  int `json:"public"`
        private int `json:"private"`
    }

With new option it would add

    type Struct struct {
        Public  int `json:"public"`
        private int
    }

### CODE REVIEW

**Comment by fatih** [comment]
Could you please more specific on what you're trying to do?  You should at least include some new tests and provide a better and readable description in the PR.

**Comment by igrmk** [comment]
I've updated the description and pushed tests

File: main.go [hunk 1]
```diff
@@ -617,13 +621,25 @@ func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {
...
+							}
+							break
+						}
+					}
+					if fieldName == "" {
+						continue
+					}
+				} else {
+					fieldName = f.Names[0].Name
+				}
```
Comment (fatih) [inline main.go]: I don't like using labels, but I think we can still achieve it by using an anonymous function. It also reads easier:

```diff
diff --git a/main.go b/main.go
index c7e0276..14b8a7b 100644
--- a/main.go
+++ b/main.go
@@ -608,6 +608,19 @@ func (c *config) offsetSelection(file ast.Node) (int, int, error) {
 func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {
        errs := &rewriteErrors{errs: make([]error, 0)}

+       processField := func(name string) bool {
+               if !c.skipPrivateFields {
+                       return true
+               }
+
+               for _, r := range name {
+                       if unicode.IsUpper(r) {
+                               return true
+                       }
+               }
+               return false
+       }
+
        rewriteFunc := func(n ast.Node) bool {
                x, ok := n.(*ast.StructType)
                if !ok {
@@ -623,22 +636,15 @@ func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {

                        fieldName := ""
                        if len(f.Names) != 0 {
-                               if c.skipPrivateFields {
-                               publicFieldSearch:
-                                       for _, field := range f.Names {
-                                               for _, c := range field.Name {
-                                                       if unicode.IsUpper(c) {
-                                                               fieldName = field.Name
-                                                               break publicFieldSearch
-                                                       }
-                                                       break
-                                               }
-                                       }
-                                       if fieldName == "" {
-                                               continue
+                               for _, field := range f.Names {
+                                       if processField(field.Name) {
+                                               fieldName = field.Name
                                        }
-                               } else {
-                                       fieldName = f.Names[0].Name
+                               }
+
+                               // nothing to process, continue with next line
+                               if fieldName == "" {
+                                       return true
                                }
                        }
```

Comment (igrmk) [reply main.go]: It seems more logical to me to extract `isPublicName` function.
Could you please review the version that I've just pushed.
Your additional comment "nothing to process, continue with next line" is incorporated.

File: main.go [hunk 2]
```diff
@@ -599,6 +603,16 @@ func (c *config) offsetSelection(file ast.Node) (int, int, error) {
...
 
+func isPublicName(name string) bool {
+	for _, c := range name {
+		if unicode.IsUpper(c) {
+			return true
+		}
+		return false
+	}
+	return false
+}
```
Comment (fatih) [inline main.go]: I think you can simplify this with the following function:

```go
func isUpper(name string) bool {
	for _, c := range name {
		return unicode.IsUpper(c)
	}
	return false
}
```

example: https://play.golang.org/p/J2SjN9V3Dhh

Comment (igrmk) [reply main.go]: Oh, such an obvious thing))) Fixed of course. Thank you!

**Comment by fatih** [comment]
This looks good 👍  Thank you for working on this @igrmk.

**Pull request merged**